test/test-basics.js

```diff
@@ -0,0 +1,46 @@
+'use strict'
+/* globals describe, it */
+const main = require('../')
+const assert = require('assert')
+const multiformats = require('multiformats/basics')
+
+multiformats.add(main)
+const { CID, multicodec, bytes } = multiformats
+const dag = {
+  encode: v => multicodec.encode(v, 'dag-json'),
+  decode: v => multicodec.decode(v, 'dag-json')
+}
+const same = assert.deepStrictEqual
+const test = it
+
+const recode = buffer => dag.encode(dag.decode(buffer))
+
+const link = new CID('bafyreifepiu23okq5zuyvyhsoiazv2icw2van3s7ko6d3ixl5jx2yj2yhu')
```

Comment:
It would make sense to also add a test that CIDv0 are bae58 encoded correctly (and also add a check that this CIDv1 is encoded as Base32, perhaps also testing with a non-base32 encoded CID)

Reply:
The latest `dag-json` spec matches the enforced defaults of the new `CID.toString()` implementation in `js-multiformats`. There are tests for this exact behavior in `js-multiformats`.

What do we think the enforced behavior should be when reading data that is encoded with a different base encoding? Should we be aggressive and actually throw in the internals when we encounter this?

Reply:
I only care about the serialization (hence interoperability with other implementations).

I see it as a blackbox test. You put in a `CID` and DAG-JSON should do the right thing. I don't care about the underlying implementation.

But even if I see it as a whitebox test, my concern would be that we depend on the default behaviour of js-multiformats. I'd like to have a test failing in case those defaults change.

Reply:
I think a better approach for these is to create some shared test fixtures for `dag-json` that any implementation can use for compliance.

Ideally, we would have the same objects in multiple formats so we can test the language’s data model representation against other’s that are known to be compliant. This would also give us a good way to test all the nasty corners of determinism in the formats against the known hashes of compliant representations

Reply:
Surely fixtures would be better, but I think having a test until we have them would also be better than nothing :)